### PR TITLE
Added Matrix Exponentiation

### PR DESCRIPTION
**Pull request by c-utkarsh**

### CODE REVIEW

File: algorithms/matrix/__init__.py [hunk 1]
```diff
@@ -1 +1 @@
-
+from .matrix_exponentiation import *
```
Comment (goswami-rahul) [inline algorithms/matrix/__init__.py:1]: We don't need this. `multiply` collides with `multiply.py`.

File: algorithms/matrix/matrix_exponentiation.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    elif n % 2 == 1:
+        return multiply(matrix_exponentiation(mat, n - 1), mat)
+    else:
+        tmp = matrix_exponentiation(mat, n // 2)
+        return multiply(tmp, tmp)
+
+if __name__ == "__main__":
+    mat = [[1, 0, 2], [2, 1, 0], [0, 2, 1]]
+    
+    res0 = matrix_exponentiation(mat, 0)
```
Comment (goswami-rahul) [inline algorithms/matrix/matrix_exponentiation.py:46]: Add tests at [`tests/test_matrix.py`](https://github.com/keon/algorithms/blob/master/tests/test_matrix.py) similar to other tests.

**Review by goswami-rahul — CHANGES_REQUESTED** [review]

**Comment by c-utkarsh** [comment]
Hi @goswami-rahul, I've incorporated the requested changes.

**Review by goswami-rahul — APPROVED** [review]
Thanks. LGTM

**Pull request merged**